command/agent/config.go

```diff
@@ -1371,6 +1376,9 @@ func MergeConfig(a, b *Config) *Config {
 	if b.NodeID != "" {
 		result.NodeID = b.NodeID
 	}
+	if b.DisableHostNodeID == true {
+		result.DisableHostNodeID = b.DisableHostNodeID
```

Comment:
Unfortunately that'll let a merge with the default value of `false` overwrite a previous value of `true`. In other words, we use the fact that it's `true` to indicate that the user actively set it. We could use a `*bool` here, but this way works as well since this defaults to `false`.